lib/src/vector_tile_provider.dart

```diff
@@ -95,6 +85,36 @@ class MemoryCacheVectorTileProvider extends VectorTileProvider {
   }
 }
 
+/// [assetPathTemplate] the asset path template, e.g. `'assets/map_tiles/{z}/{x}/{y}.pbf'`
+/// [maximumZoom] the maximum zoom supported by the tile provider, not to be
+/// confused with the maximum zoom of the map widget. The map widget will
+/// automatically use vector tiles from lower zoom levels once the maximum
+/// supported by this provider is reached.
+class AssetVectorTileProvider extends VectorTileProvider {
+  AssetVectorTileProvider({required this.assetPathTemplate, required this.maximumZoom});
+  @override
+  final int maximumZoom;
+  final String assetPathTemplate;
+
+  @override
+  Future<Uint8List> provide(TileIdentity tile) {
+    final path = assetPathTemplate.replaceAllMapped(RegExp(r'\{(x|y|z)\}'), (match) {
+      switch (match.group(1)) {
+        case 'x':
+          return tile.x.toString();
+        case 'y':
+          return tile.y.toString();
+        case 'z':
+          return tile.z.toString();
+        default:
+          throw Exception('unexpected url template: $assetPathTemplate - token ${match.group(1)} is not supported');
```

Comment:
looks like some code duplication here since `_UrlProvider` does the same thing. Would it make sense to rename `_UrlProvider` to `_UriProvider` and reuse it for both cases?